components/automate-ui/src/app/entities/rules/rule.model.ts

```diff
@@ -1,14 +1,28 @@
+export type RuleStatus = 'applied' | 'staged';
+export type RuleType = 'node' | 'event';
+export type ConditionOperator = 'MEMBER_OF' | 'EQUALS';
+
+export interface KVPair {
+  key: string;
+  value: string;
+}
+
+// Using array of pairs instead of a hashmap to allow ordering
+export type RuleTypeMappedObject = {
+  [K in 'node' | 'event']: KVPair[]
+};
```

Comment:
Tightened up type constraints a lot, with the help of these. 

The first few are string literal types. (Thanks @srenatus for reminding me about those!)

The last one is a `mapped type` that allows defining a complex typed constant in rule.reducer.ts. (More info: http://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types)